core/src/main/scala/chisel3/properties/Property.scala

```diff
@@ -307,24 +263,19 @@ abstract class Property[T] extends BaseType { self =>
   */
 object Property {
 
-  private[chisel3] def makeWithValueOpt[T](valueOpt: Option[T])(implicit _tpe: PropertyType[T]): Property[_tpe.Type] = {
-    new Property[_tpe.Type] {
-      type TT = T
-      val tpe = _tpe
-      val value = valueOpt
-    }
-  }
+  private[chisel3] def makeWithValueOpt[T](valueOpt: Option[T])(implicit _tpe: PropertyType[T]): Property[T] =
+    new Property[T](_tpe, valueOpt)
 
   /** Create a new Property based on the type T.
     */
-  def apply[T]()(implicit tpe: PropertyType[T]): Property[tpe.Type] = {
+  def apply[T]()(implicit tpe: PropertyType[T]): Property[T] = {
```

Comment:
The path dependent type was necessary before so that e.g. `IO(Output(Property[Int])) := Property(Property[Int]())` would work since `:=` only worked for `Property[T]` with the same type parameter. Now that `:=` works to connect any `Property[_]` regardless of type parametrization, it is not necessary for the return types to be dependent.

But the return type is still important for users that want to layer extra type safety on top of this API e.g. 
```scala
def connectProp(prop: Property[Seq[Int]]) =
  io := prop

// this would not type check anymore
connectProp(Property(Seq(Property[Int](123), IO(Input(Property[Int]())))
```

So I think we still want to keep the return type as `Property[tpe.Type]`, or just get rid of the type parameter entirely `Property[T]` -> `Property` since the core operations on `Property` aren't parameterized on `T` anyway.

Reply:
Oh I understand now. Yeah even though `:=` works for any `Property` (because it works for any `Data`), the connectable operators are still typesafe, so I can test for this behavior by using `:#=` instead of `:=` connecting a `Property[Seq[Property[A]]]` to a `Property[Seq[A]]`.

Thanks for the explanation! I will restore the path dependent type.